piqueserver/scripts/savemap.py

```diff
@@ -8,67 +14,68 @@
+def rmsaved(connection):
+    name = connection.protocol.map_info.rot_info.name
+    path = get_path(name)
+    if os.path.isfile(path):
+        os.remove(path)
+        # remove .saved suffix
+        connection.protocol.map_info.rot_info.name = name[:-6]
+        return "Map '%s' removed" % name
+    else:
+        return "There is no saved version of '%s' map" % name
+
+def get_path(map_name):
+    if map_name.endswith('.saved'):
+        map_name = map_name[:-6]
+    return '%s.saved.vxl' % os.path.join(config_dir, 'maps', map_name)
 
 def apply_script(protocol, connection, config):
-    class MapSaveProtocol(protocol):
-
+    class SaveMapProtocol(protocol):
         def __init__(self, *arg, **kw):
             protocol.__init__(self, *arg, **kw)
-            reactor.addSystemEventTrigger('before', 'shutdown', self.save_map)
-
-        def make_map(self, name):
-            if not LOAD_SAVED_MAP_OPTION.get():
-                # just do the normal operation
-                return protocol.make_map(self, name)
-
-            async def do_load():
-                map_info = await protocol.make_map(self, name)
-                cached_path = get_name(map_info)
-
-                if not os.path.isfile(cached_path):
-                    # no cache file. Just load the regular map
-                    return map_info
-
-                log.info("Loading saved map for {name} from {cached_path}",
-                    name=name, cached_path=cached_path)
-                map_info.data = await threads.deferToThread(
-                    load_map_from_path, cached_path)
-                log.info("Saved map loaded")
-                return map_info
-
-            # replace the map_info deferred with our own one that also
-            # loads and replaces the map data
-            return ensureDeferred(do_load)
-
-        def save_map(self):
-            with open(get_name(self.map_info), 'wb') as f:
+            def call():
+                if savemap_config.option('save_at_shutdown', False).get():
+                    self.save_map()
+            reactor.addSystemEventTrigger('before', 'shutdown', call)
+
+        async def set_map_name(self, rot_info: RotationInfo) -> None:
+            if savemap_config.option('load_saved_map', False).get():
+                if os.path.isfile(get_path(rot_info.name)):
+                    log.info("Saved version of '%s' found" % rot_info.name)
+                    rot_info.name += '.saved'
+            await protocol.set_map_name(self, rot_info)
+
+        def save_map(self, custom_name=None):
+            if custom_name:
+                path = os.path.join(config_dir, 'maps', custom_name) + '.vxl'
+            else:
+                path = get_path(self.map_info.rot_info.name)
+            with open(path, 'wb') as f:
                 f.write(self.map.generate())
-            log.info("Map saved to {}".format(get_name(self.map_info)))
+            log.info("Map saved to '%s'" % path)
```

Comment:
```suggestion
            log.info("Map saved to '{path}'", path=path)
```

Reply:
And for what?

Reply:
This is just the correct way of using the logging api. The formatting code won't be run unless the message is actually printed and you can emit the logs as easily parseable JSON if you want.